### PR TITLE
Feature/minor fixes

### PR DESCRIPTION
**Pull request by ice-hector**

### CODE REVIEW

File: src/screens/HomeFlow/Home/components/Header/components/UserGreeting.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export function UserGreeting({disabled, animatedStyle}: Props) {
           <GreetingText />
           {user && (
```
Comment (ice-hades) [inline src/screens/HomeFlow/Home/components/Header/components/UserGreeting.tsx]: user?.username &&

Comment (ice-hades) [reply src/screens/HomeFlow/Home/components/Header/components/UserGreeting.tsx]: or we should render an empty string even if no username?

File: src/screens/ProfileFlow/Profile/components/AvatarHeader/index.tsx [hunk 2]
```diff
@@ -113,7 +113,7 @@ export const AvatarHeader = memo(
               <Animated.Text
                 style={[styles.usernameText, textStyle]}
                 numberOfLines={1}>
-                {buildUsernameWithPrefix(user.username)}
+                {buildUsernameWithPrefix(user.username ?? '')}
```
Comment (ice-hades) [inline src/screens/ProfileFlow/Profile/components/AvatarHeader/index.tsx]: same here. maybe better to check user?.username on the render condition?

File: src/navigation/Main.tsx [hunk 3]
```diff
@@ -74,6 +74,8 @@ import {SvgProps} from 'react-native-svg';
 import {useDispatch} from 'react-redux';
 import {rem} from 'rn-units/index';
 
+type ProfileTabActionType = 'updatePhoto';
```
Comment (ice-orion) [inline src/navigation/Main.tsx]: Let's please keep a screen related types in a screen (like e.g. for PopUp -> PopUpProps or ContextualMenu->ContextualMenuButton).

And why `Tab` in the name `ProfileTabActionType`? Probably just `ProfileActionType`?

File: src/screens/ProfileFlow/Profile/components/Role/components/CurrentRoleCard.tsx [hunk 4]
```diff
@@ -38,7 +38,8 @@ export const CurrentRoleCard = ({
   const {onUpdate} = useUpdateHiddenProfileElements();
   const isPrivacyInfoShown = useSelector(isPrivacyInfoShownSelector);
 
-  const isRoleHidden = user?.hiddenProfileElements?.includes('role');
+  const isRoleHidden: boolean =
```
Comment (ice-orion) [inline src/screens/ProfileFlow/Profile/components/Role/components/CurrentRoleCard.tsx]: Not critical, but no need to set the type explicitly here, ts will infer it anyway:

```
const isRoleHidden = !!user?.hiddenProfileElements?.includes('role');
```

File: src/screens/ProfileFlow/Profile/hooks/useUserData.ts [hunk 5]
```diff
@@ -1,27 +1,49 @@
...
+
   const isUserLoading = useSelector(
     isLoadingSelector.bind(null, UsersActions.GET_USER_BY_ID),
   );
 
+  useEffect(() => {
+    if (route.params?.actionType === 'updatePhoto' && !actionSheetShown) {
+      setTimeout(() => {
+        onEditPress();
+        setActionSheetShown(true);
```
Comment (ice-orion) [inline src/screens/ProfileFlow/Profile/hooks/useUserData.ts]: I would say this is not the best way to process the router params here - 
This hook is about fetching user data, not about dealing with the action sheet and avatar
And calling `useUpdateAvatar` you ignore the loading indicator that is also returned from `useUpdateAvatar` and should be applied to UI.

What if we create a hook in `src/screens/ProfileFlow/Profile/components/AvatarHeader/hooks`, e.g. `useUpdateAvatarRouterParam` that would accept and call the same function that is in `EditAvatarButton`->`onPress` when `route.params?.actionType === 'updatePhoto' && !actionSheetShown` ?

File: src/screens/HomeFlow/Home/components/Header/components/UserGreeting.tsx (hunk 1 above)
Comment (ice-hector) [reply src/screens/HomeFlow/Home/components/Header/components/UserGreeting.tsx]: Changed to user.username

File: src/screens/ProfileFlow/Profile/components/AvatarHeader/index.tsx (hunk 2 above)
Comment (ice-hector) [reply src/screens/ProfileFlow/Profile/components/AvatarHeader/index.tsx]: Added default value empty string inside the `buildUsernameWithPrefix `

**Review by ice-orion — APPROVED** [review]

**Review by ice-atlas — APPROVED** [review]

**Review by ice-atlas — APPROVED** [review]

**Review by ice-hades — APPROVED** [review]

**Review by ice-atlas — APPROVED** [review]

**Pull request merged**